zebra-state/src/service/non_finalized_state/tests/prop.rs

```diff
@@ -129,73 +135,90 @@ fn forked_equals_pushed_genesis() -> Result<()> {
+            empty_tree.clone(),
+            ValueBalance::zero(),
+        );
         for block in chain.iter().cloned() {
             let block =
-                ContextuallyValidBlock::with_block_and_spent_utxos(
-                    block,
-                    full_chain.unspent_utxos(),
-                )?;
-                full_chain = full_chain.push(block.clone()).expect("full chain push is valid");
-
-                // check some other properties of generated chains
-                if block.height == block::Height(0) {
-                    prop_assert_eq!(
-                        block
-                            .block
-                            .transactions
-                            .iter()
-                            .flat_map(|t| t.inputs())
-                            .filter_map(|i| i.outpoint())
-                            .count(),
-                        0,
-                        "unexpected transparent prevout input at height {:?}: \
-                         genesis transparent outputs must be ignored, \
-                         so there can not be any spends in the genesis block",
-                        block.height,
-                    );
-                }
+                ContextuallyValidBlock::with_block_and_spent_utxos(block, full_chain.unspent_utxos())?;
+            full_chain = full_chain
+                .push(block.clone())
+                .expect("full chain push is valid");
+
+            // Check some other properties of generated chains.
+            if block.height == block::Height(0) {
+                prop_assert_eq!(
+                    block
+                        .block
+                        .transactions
+                        .iter()
+                        .flat_map(|t| t.inputs())
+                        .filter_map(|i| i.outpoint())
+                        .count(),
+                    0,
+                    "unexpected transparent prevout input at height {:?}: \
+                            genesis transparent outputs must be ignored, \
+                            so there can not be any spends in the genesis block",
+                    block.height,
+                );
             }
+        }
 
+        // Use [`fork_at_count`] as the fork tip.
+        let fork_tip_hash = chain[fork_at_count - 1].hash;
+
+        // Fork the chain.
         let mut forked = full_chain
             .fork(
                 fork_tip_hash,
                 Default::default(),
                 Default::default(),
+                Default::default(),
                 empty_tree,
             )
             .expect("fork works")
             .expect("hash is present");
 
-        // the first check is redundant, but it's useful for debugging
+        // This check is redundant, but it's useful for debugging.
         prop_assert_eq!(forked.blocks.len(), partial_chain.blocks.len());
+
+        // Check that the blocks in the forked chain correspond to the blocks in
```

Comment:
Comment Fix: It's more than just the blocks, it's the derived data as well.

```suggestion
        // Check that the entire internal state of the forked chain corresponds to the state of
```

Reply:
I thought of this as well, but I couldn't come up with a suitable phrasing. This looks much better.